readme.md

```diff
@@ -59,6 +59,25 @@ Whether to run the check immediately rather than starting by waiting `interval`
 
 Useful for when the check, if run immediately, would likely return `false`. In this scenario, set `before` to `false`.
 
+#### resolveWith(value)
+
+Resolve the main promise with a custom value.
+
+```js
+import pWaitFor from 'p-wait-for';
+import pathExists from 'path-exists';
+
+const path = await pWaitFor(async resolveWith => {
```

Comment:
Any thoughts about having it as a parameter like this vs. a standalone function like `pWaitFor.resolveWith()`? I'm honestly not sure which is better.

Reply:
I think a standalone function might be better because it's a bit more obvious that the value is meant to be returned from the function, whereas it's less clear with the `resolveWith` parameter since it's similar to the `resolve` parameter in Promises which doesn't need to be returned.

Reply:
@leonzalion That's a good argument.

Reply:
Then the question is whether it should be `pWaitFor.resolveWith()` or a named export. And if a named export, what should the name be so it's not ambiguous.

Reply:
I imagine it's going to be pretty hard to think of a non-ambiguous named export that isn't awkward to type; I think `pWaitFor.resolveWith()` is ok, similar to `onetime.callCount()` (https://github.com/sindresorhus/onetime#onetimecallcountfn)

Reply:
The whole point of having it like a parameter is so that it is easy to write the typings for it. Though, on a second look, it seems even easier.